client/templates/userprofile/settings.js

```diff
@@ -1,151 +1,59 @@
-        return data = ["English", "Finnish", "German", "Persian", "Russian", "Swedish"];
-    }
-});
-
-Template.profileSettings.events({
-
-    'change #realName': function(event,template) {
-      if(template.find("#realName").value)
-      {
-          Session.set("realerror", "");
-      }
-        else
-      {
-          Session.set("realerror", "Your Realname is empty!");
-      }
-    },
-    'change #userName': function(event,template) {
-        if(template.find("#userName").value)
-        {
-            Session.set("usererror", "");
-        }
-        else
-        {
-            Session.set("usererror", "Your username is empty!");
-        }
-    },
-    'change #email': function(event,template) {
-        if(template.find("#email").value)
-        {
-            Session.set("emailerror", "");
-            if(!(isEmail(template.find("#email".value))))
+    genderOptions: function() {
+        return [
             {
-                Session.set("emailerror", "You have not entered a correct Email!");
+                optgroup: "Please select your Gender",
+                options: [
+                    {label: "Female", value: "Female"},
+                    {label: "Male", value: "Male"},
+                    {label: "I'd rather not tell", value: "Unknown"}
+                ]
             }
-            else
+        ]
+    },
+    languageOptions: function() {
+        return [
             {
-                Session.set("emailerror", "");
+                optgroup: "Please select your preferred Language",
+                options: [
+                    {label: "English", value: "English"},
+                    {label: "Finnish", value: "Finnish"},
+                    {label: "German", value: "German"},
+                    {label: "Russian", value: "Russian"},
+                    {label: "Persian", value: "Persian"}
+
+                ]
             }
-        }
-        else
-        {
-            Session.set("emailerror", "Your Email is empty!");
-        }
-
-    },
+        ]
+    }
 
-    'click #saveChanges': function(event, template) {
-        var realname = template.find("#realName").value;
-        var username = template.find("#userName").value;
-        var gender = template.find("#gender option:selected").value;
-        var language = template.find("#language option:selected").value;
-        var biography = template.find("#biography").value;
-        var email = template.find("#email").value;
+});
 
-        if (Meteor.userId())
-        {
-            if (realname && username && gender && language && email && isEmail(email))
+Template.profileSettings.events({
+    "submit #passwordEdit": function (event) {
+        event.preventDefault();
+        //TODO Unshow error message after a while
+        oldPassword = AutoForm.getFieldValue('old', 'passwordEdit');
+        newPassword = AutoForm.getFieldValue('new', 'passwordEdit');
+        Accounts.changePassword(oldPassword, newPassword, function(error) {
```

Comment:
It might be better to move this into an AutoForm callback. You can copy and paste this code, so no new work is necessary. For reference, see this [exact implementation in Apinf code](https://github.com/apinf/api-umbrella-dashboard/blob/8fba1276c2ad02b1d41f17de57a0b6d3a7abf48d/client/views/account/account.js#L2-L22).